specification/specs/x_pack/machine_learning/post_job_data/PostJobDataRequest.ts

```diff
@@ -31,7 +31,6 @@ interface PostJobDataRequest extends RequestBase {
     reset_end?: DateString
     reset_start?: DateString
   }
-  body?: {
-    data?: UserDefinedValue[]
-  }
+  // TODO: should be name JSON Document
+  body: UserDefinedValue
```

Comment:
According to the doc and the rest-api-spec, this API expects a bulk body, we should express it in the same way of the [BulkRequest.ts](https://github.com/elastic/elastic-client-generator/blob/master/specification/specs/document/multiple/bulk/BulkRequest.ts).

```ts
interface PostJobDataRequest<TSource> extends RequestBase {
  path_parts?: {
    job_id: Id
  }
  query_parameters?: {
    reset_end?: DateString
    reset_start?: DateString
  }
  body?: TSource[]
}
```

Also, we are not signaling in any way in the input spec that an API requires a ndjson body, we should do it. I'll open a separate issue for this.

Reply:
https://github.com/elastic/elastic-client-generator/issues/297

Reply:
Added a generic as body, but it must be a json not ndjson, see [docs](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/ml-post-data.html#_request_body_23)

Reply:
From the docs:

> A sequence of one or more JSON documents containing the data to be analyzed. Only whitespace characters are permitted in between the documents.

The rest-api-spec for this endpoint:

```jsoni
    "body":{
      "description":"The data to process",
      "required":true,
      "serialize":"bulk"
    }
```